### PR TITLE
[COLLECTIONS-739] Fix inconsistent @throws comments in DefaultedMap

### PR DESCRIPTION
**Pull request by Prodigysov**
See [COLLECTIONS-739](https://issues.apache.org/jira/browse/COLLECTIONS-739).

### CODE REVIEW

File: src/main/java/org/apache/commons/collections4/map/DefaultedMap.java [hunk 1]
```diff
@@ -105,7 +105,8 @@
      * @param map  the map to decorate, must not be null
      * @param factory  the factory to use to create entries, must not be null
      * @return a new defaulting map
-     * @throws NullPointerException if map or factory is null
+     * @throws NullPointerException if map is null
+     * @throws IllegalArgumentException if factory is null
```
Comment (dota17) [inline src/main/java/org/apache/commons/collections4/map/DefaultedMap.java]: This is indeed a judgment of NullPointerException. It is recommended to modify IllegalArgumentException to NullPointerException, and to supplement the test cases for it.

Comment (Prodigysov) [reply src/main/java/org/apache/commons/collections4/map/DefaultedMap.java]: Agreed. I changed to NullPointerException and added test cases for it.

File: src/main/java/org/apache/commons/collections4/map/DefaultedMap.java [hunk 2]
```diff
@@ -110,7 +110,7 @@
      */
     public static <K, V> DefaultedMap<K, V> defaultedMap(final Map<K, V> map, final Factory<? extends V> factory) {
         if (factory == null) {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/collections4/map/DefaultedMap.java]: Not quite, to go the NPE route, you should use a liner with `Objects.requireNonNull()`.

Comment (Prodigysov) [reply src/main/java/org/apache/commons/collections4/map/DefaultedMap.java]: Changed to use `Objects.requireNonNull` (I noticed and followed commit 513ae545ee2b333f62f8ff989e1da161d807d76d)

**Pull request merged**